checked/ref.go

```diff
@@ -169,3 +171,37 @@ func (c *RefCount) DecWrites() {
 func (c *RefCount) NumWriters() int {
 	return int(atomic.LoadInt32(&c.writes))
 }
+
+// TrackObject sets up the initial internal state of the Ref for
+// leak detection.
+func (c *RefCount) TrackObject(v interface{}) {
+	if atomic.LoadUint64(&leakDetectionFlag) == 0 {
+		return
+	}
```

Comment:
Could export a method `ShouldTrackObject() bool` that performs this so others outside the module can use it?